truss/templates/server/common/truss_server.py

```diff
@@ -255,6 +256,20 @@ def create_application(self):
             },
         )
 
+        async def dispatch(request, call_next):
+            request_id = request.headers.get("x-baseten-request-id", uuid.uuid4())
+            with loguru_logger.contextualize(request_id=request_id):
+                try:
+                    return await call_next(request)
+
+                except Exception as ex:
+                    print(f"Request failed: {ex}")
+                    return JSONResponse(content={"success": False}, status_code=500)
+
+                finally:
+                    # sentinel log
+                    print("Request complete")
```

Comment:
will this be visible to users? what is the point of this?

Reply:
We need a log after the request completes so we can group all requests with the same `id` from the front-end as a single grouping. This means we either need a sentinel log like this one or we override the POST log to be tagged with a `request_id`. I'm looking into how to do the latter.